ts/Core/Utilities.ts

```diff
@@ -462,6 +462,29 @@ function erase(arr: Array<unknown>, item: unknown): void {
     }
 }
 
+/**
+ * Adds an item to an array, if it is not present in the array.
+ *
+ * @param {Array<unknown>} array
+ * The array to add the item to.
+ *
+ * @param {unknown} item
+ * The item to add.
+ *
+ * @return {boolean}
+ * Returns true, if the item was not present and has been added.
+ */
+function pushUnique(
+    array: Array<unknown>,
+    item: unknown
+): boolean {
+    if (array.indexOf(item) === -1) {
+        array.push(item);
+        return true;
+    }
+    return false;
```

Comment:
Maybe we can save a few bytes by typing the function `true|undefined` and skip this line?